components/kyma-environment-broker/internal/error/error.go

```diff
@@ -0,0 +1,38 @@
+package error
+
+import "fmt"
+
+type TemporaryError struct {
+	message string
+}
+
+func NewTemporaryError(message string, err error) *TemporaryError {
+	var msg string
+	if err != nil {
+		msg = fmt.Sprintf("%s: %s", message, err.Error())
+	} else {
+		msg = fmt.Sprintf("%s", message)
+	}
+
+	return &TemporaryError{message: msg}
+}
+
+func (te TemporaryError) Error() string { return te.message }
+func (TemporaryError) Temporary() bool  { return true }
+
+func IsTemporaryError(err error) bool {
+	nfe, ok := err.(interface {
+		Temporary() bool
+	})
+	return ok && nfe.Temporary()
+}
+
+func Wrapf(err error, format string, args ...interface{}) error {
```

Comment:
IMO we do not need that function. If you want to have such behavior, then just modify the `IsTemporaryError` func.

```go
func IsTemporaryError(err error) bool {
	cause := errors.Cause(err)
	nfe, ok := cause.(interface {
		Temporary() bool
	})
	return ok && nfe.Temporary()
}
```
In this approach is just 1 line change and it is more generic. 

This pattern was also described by Dave Cheney, the guy who created /pkg/errors lib. 
_Source: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully_ 

Additionally when we're introducing the `Wrapf` func then we need to remember to always use it in our code, but we used to just use`errors.Wrap` and `errors.Wrapf`. Also you sometimes you need to have such statements which you added:
```
		// do not wrap error, TemporaryError type can be return
```

Reply:
perfect, I was looking for that